pkg/rewards/10_goldAvsODRewardAmounts.go

```diff
@@ -0,0 +1,78 @@
+package rewards
+
+import (
+	"github.com/Layr-Labs/sidecar/pkg/rewardsUtils"
+	"go.uber.org/zap"
+)
+
+const _10_goldAvsODRewardAmountsQuery = `
+create table {{.destTableName}} as
+-- Get the rows where operators have not registered for the AVS or if the AVS does not exist
+WITH reward_snapshot_operators as (
+  SELECT
+    ap.reward_hash,
+    ap.snapshot as snapshot,
+    ap.token,
+    ap.tokens_per_day,
+    ap.tokens_per_day_decimal,
+    ap.avs as avs,
+    ap.operator as operator,
+    ap.strategy,
+    ap.multiplier,
+    ap.reward_submission_date
+  FROM {{.activeRewardsTable}} ap
+  LEFT JOIN operator_avs_registration_snapshots oar
+  ON ap.avs = oar.avs and ap.snapshot = oar.snapshot and ap.operator = oar.operator
+  WHERE oar.avs IS NULL OR oar.operator IS NULL
+),
+-- Dedupe the operator tokens across strategies for each (operator, reward hash, snapshot) since the above result is a flattened out operator-directed reward submission across strategies.
+distinct_operators AS (
+  SELECT *
+  FROM (
+	  SELECT *,
+	    -- We can use an arbitrary order here since the avs_tokens is the same for each (operator, strategy, hash, snapshot)
+        -- We use strategy ASC for better debuggability
+		ROW_NUMBER() OVER (PARTITION BY reward_hash, snapshot, operator ORDER BY strategy ASC) as rn
+	  FROM reward_snapshot_operators
+  ) t
+  WHERE rn = 1
+),
+-- Sum the operator tokens for each (reward hash, snapshot), since we want to refund the sum of those operator amounts to the AVS in that reward submission for that snapshot.
+operator_token_sums AS (
+  SELECT
+	reward_hash,
+	snapshot,
+	token,
+	avs,
+	operator,
+	SUM(tokens_per_day_decimal) OVER (PARTITION BY reward_hash, snapshot) AS avs_tokens
+  FROM distinct_operators
+)
+SELECT * FROM operator_token_sums
```

Comment:
Can we add an invariant check (unsure if we are adding these in the sidecar, but we should) where for a given snapshot:

sum(all_operators_staker_tokens) + operator_tokens = total_staker_operator_payout. 

avs_tokens + sum(total_staker_operator_payout) = avs_tokens_per_day

Reply:
@seanmcgary Do we do this anywhere currently in the sidecar (like for regular rewards)?

Reply:
@ypatil12 I could have a 5th SQL step checking the above invariant. Does that work? Or maybe even in just Go Code. How did we have invariant checks in the pipeline?

Reply:
Current pipeline has a DAG step that runs this module: https://github.com/Layr-Labs/block-pipe/tree/preprod/src/modules/gold_table_sanity_checker

Reply:
I've noted this down as a TODO for a quick follow-on PR so that we can keep this PR in scope.